ui/src/main/java/org/pentaho/di/ui/job/entry/JobEntryDialog.java

```diff
@@ -303,7 +309,19 @@ String showDbDialogUnlessCancelledOrValid( DatabaseMeta changing, DatabaseMeta o
           // OK was pressed and input is valid
           repeat = false;
         } else {
-          showDbExistsDialog( changing );
+          try {
+            DatabaseManagementInterface dbMgr =
+              spoonSupplier.get().getBowl().getManager( DatabaseManagementInterface.class );
```

Comment:
This is correct for new connections, but this function is also used for Edit. The DatabaseManagementInterface may need to be passed into this function for the different use cases.

Reply:
done